---
title: "Problem Set 3. Due Friday March 11 5pm"
author: "Student name"
date: "March 7, 2016"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, comment = "#>")
```

# Submission

1. Fork this repository to your own account
1. Make changes to the `README.Rmd` file (including the author field above).
1. Knit the file to HTML format
1. Publish HTML on Rpubs (you will need to sign up for the free account) and *add the link below*.
1. Submit a PR with your answers.

*Link to Rpubs document:* XXX

# Overview 

We will examine a [time-course of mRNA expression][1] of barley powdery mildew fungus in immunocompromised plants (*Arabidopsis*). The fungus Blumeria graminis f. sp. hordei (called *Bgh* from now on)

Counts were generated using the [`maSigPro` software][3] and deposited in [NCBI GEO][2].

[1]: http://www.ncbi.nlm.nih.gov/pubmed/23175786
[2]: http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE43163  
[3]: http://bioinformatics.oxfordjournals.org/content/30/18/2598.full

Some important identifiers for the data set:

| Name | Description |
| ---- | ----------- |
| `A6` | *Bgh* isolate expressing other AVRA effectors |
| `K1` | *Bgh* isolate expressing the cognate AVRA1 effector for MLA1|
| `pps` | *Arabidopsis* plants |
| `B12` | *Arabidopsis* plants expressing MLA1-HA|

We will download tables from GEO containing counts of mRNA abundance from both the fungal sample.

# Raw data

First we need load the data.

```{r load_raw_data, message = FALSE}
library(readr)
library(tidyr)
library(dplyr)
library(stringr)

# bgh data
bgh_url <- 'http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE43163&format=file&file=GSE43163_CompleteCountTable_Bgh.txt.gz'

# 2 info and blank lines at top, skip them
raw_data <- read_tsv(bgh_url, skip = 2)

# the header for the first column is "NA", set it manually
names(raw_data)[1] <- 'gene.name'

raw_data
```

# Problems

## Problem 1

Now that the raw data are loaded, your first task is to tidy the data with `tidyr`.

```{r problem_1}
# add tidying code here
cleaned_data <- raw_data %>% tbl_df

cleaned_data
```

## Problem 2

Identify the top 3 most consistently differentially expressed genes between the earliest and latest time points for each combination of `plant` and `Bgh` strains.

- Differential expression is the difference between `counts`
- "Consistency" is the smallest variance in counts between replicates

### Strategy

1. Create a new table from the cleaned data by moving each `hpi` value to a new column name, with counts for each in the column (hint: use a `tidyr` verb).

2. Create a new column containing the expression difference between the relevant time points.

3. Calculate summary statistics (mean and variance) of the expression differences by grouping (hint) the `gene.name`, `plant`, and `virus` columns.

4. Sort by these statistics and use the `dplyr` verb `slice` to pull the ones you want (i.e., the top 3). Note you will have to remove `gene.name` from the grouping so that sorting works.

```{r problem_2}
```

There are `r nrows(final_data)` rows of data in the final `data.frame`.

## Problem 3

Now examine the above final data frame above and write a few sentences putting inline code in least 3 places. For example, There are `r nrows(final_data)` rows of data in the final `data.frame`.

## Problem 4

Plot the expression (`count`) by time (`hpi`) for the above genes. Format the plot as follows:

1. Plot each point.
1. Connect the points with a smooth line 
1. Create a facet for each `plant` / `bgh` pair 
1. Assign a different color to each gene 
1. Assign a different shape to each replicate 
1. Add a descriptive title to the plot 
1. Use the "black & white" theme

Extra credit: add error bars to the plot (use `geom_errorbar`).

```{r problem_4}
library(ggplot2)
```

## Problem 5

Post your knit HTML on rpubs.com and add the link to the block at the top of the document.

